src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -168,34 +177,90 @@ public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToJSON(DataType targetCQ
           JSONCodec.ToCQL.safeNumber(BigDecimal::floatValue),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  public static final JSONCodec<BigDecimal, Integer> INT =
+  public static final JSONCodec<BigDecimal, Integer> INT_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.INT,
           JSONCodec.ToCQL.safeNumber(BigDecimal::intValueExact),
           JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
 
-  public static final JSONCodec<BigDecimal, Short> SMALLINT =
+  public static final JSONCodec<BigInteger, Integer> INT_FROM_BIG_INTEGER =
+      new JSONCodec<>(
+          GenericType.BIG_INTEGER,
+          DataTypes.INT,
+          JSONCodec.ToCQL.safeNumber(BigInteger::intValueExact),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  public static final JSONCodec<Long, Integer> INT_FROM_LONG =
+      new JSONCodec<>(
+          GenericType.LONG,
+          DataTypes.INT,
+          JSONCodec.ToCQL.safeNumber(JSONCodec.ToCQL::safeLongToInt),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  public static final JSONCodec<BigDecimal, Short> SMALLINT_FROM_BIG_DECIMAL =
```

Comment:
is this correct that we have to smallint from BigDecimal - i thought all integral would come from BigInteger ?

Reply:
This is needed in cases where JSON has, say, `1.0` and we want to allow that: Jackson bases type of textual representation of JSON Number so if there's decimal point, it's FP.

I am assuming we want to allow this coercion (existing code, when only using `BigDecimal`, did allow this): if not, could drop this converter.